tests/test_salesforce_incremental_table_reset.py

```diff
@@ -0,0 +1,145 @@
+import unittest
+import datetime
+import dateutil.parser
+import pytz
+
+from tap_tester import runner, menagerie, connections
+
+from base import SalesforceBaseTest
+
+
+class SalesforceIncrementalTableReset(SalesforceBaseTest):
+    @staticmethod
+    def name():
+        return "tap_tester_salesforce_incremental_table_reset"
+
+    @staticmethod
+    def expected_sync_streams():
+        return {
+            'Account',
+            'Contact',
+            'User',
+        }
+
+    @staticmethod
+    def convert_state_to_utc(date_str):
+        """
+        Convert a saved bookmark value of form '2020-08-25T13:17:36-07:00' to
+        a string formatted utc datetime,
+        in order to compare aginast json formatted datetime values
+        """
+        date_object = dateutil.parser.parse(date_str)
+        date_object_utc = date_object.astimezone(tz=pytz.UTC)
+        return datetime.datetime.strftime(date_object_utc, "%Y-%m-%dT%H:%M:%SZ")
+
+    def calculated_states_by_stream(self, current_state):
+        """
+        Returns the streams from the current state
+        """
+
+        stream_to_current_state = {stream : bookmark.get(self.expected_replication_keys()[stream].pop())
+                                   for stream, bookmark in current_state['bookmarks'].items()}
+        return stream_to_current_state
```

Comment:
I don't think I'm following this test method.  The name implies a calculation is being done?

Reply:
Changed the method name to "get_states_by_stream" to be reflective of what the method actually does.

Reply:
Removed the method and making the format change in the main method